app/controllers/concerns/allowed_actions.rb

```diff
@@ -0,0 +1,16 @@
+module AllowedActions
+  extend ActiveSupport::Concern
+
+  def bootdisk_type_allowed?(action = params[:action])
+    return true if Setting.find_by(name: 'bootdisk_allowed_types').value.include?(action.to_sym)
```

Comment:
Any particular reason not to use Setting["XYZ"] here? Also you can create a helper method in the Setting class named with the same name as here and call it.

Reply:
This resolves to strings here:

```
[1] pry(main)> Setting.find_by(name: 'bootdisk_allowed_types').value
=> ["generic", "host", "full_host", "subnet"]
```

I think our settings might not even support symbols to store data. My previous comment still applies too.